nsenter/event.go

```diff
@@ -742,34 +742,62 @@ func (e *NSenterEvent) processOpenFileRequest() error {
 }
 
 func (e *NSenterEvent) processFileReadRequest() error {
+	var (
+		fd  *os.File
+		err error
+		sz  int
+	)
 
 	payload := e.ReqMsg.Payload.(domain.ReadFilePayload)
 
-	// Perform read operation and return error msg should this one fail.
-	fileContent, err := ioutil.ReadFile(payload.File)
+	fd, err = os.Open(payload.File)
 	if err != nil {
 		e.ResMsg = &domain.NSenterMessage{
 			Type:    domain.ErrorResponse,
 			Payload: &fuse.IOerror{RcvError: err},
 		}
 		return nil
 	}
+	defer fd.Close()
+
+	data := make([]byte, payload.Len)
+
+	sz, err = fd.ReadAt(data, payload.Offset)
```

Comment:
That's a good point. In practice it does not make a difference, because the read() syscall usually (if not always) reads the entire proc file in one shot. 

I did it this way because I wanted to keep consistency in the code between how it handles read() and write(), and for write() we do need to only write the specified number of bytes at the specified offset. 

But note the following: changing read() to read the entire file on the backend may also be counter-productive. What if the caller wants to read only the first 10 bytes of very large file? Then he would pay the penalty for reading the entire large file on the nsenter and caching it, when we only needed the first 10 bytes. 

Of course, there is the other case you are thinking about: what if the caller reads the entire file in chunks, and we pay the price to go into the nsenter backend for each chunk. In that case reading from the backend once makes sense.

Since we don't know which to optimize for, and since read() syscall usually reads the entire proc file in one shot, I think it's fine to leave it as is.

Reply:
Fair enough.